code/modules/mob/living/simple_animal/hostile/floorcluwne.dm

```diff
@@ -0,0 +1,411 @@
+#define STAGE_HAUNT 1
+#define STAGE_SPOOK 2
+#define STAGE_TORMENT 3
+#define STAGE_ATTACK 4
+#define MANIFEST_DELAY 9
+
+/mob/living/simple_animal/hostile/floor_cluwne
+	name = "???"
+	desc = "...."
+	icon = 'icons/goonstation/objects/clothing/mask.dmi'
+	icon_state = "cursedclown"
+	icon_living = "cursedclown"
+	icon_gib = "clown_gib"
+	maxHealth = 250
+	health = 250
+	speed = -1
+	attacktext = "attacks"
+	attack_sound = 'sound/items/bikehorn.ogg'
+	del_on_death = TRUE
+	pass_flags = PASSTABLE | PASSGRILLE | PASSMOB | LETPASSTHROW | PASSGLASS | PASSBLOB//it's practically a ghost when unmanifested (under the floor)
+	loot = list(/obj/item/clothing/mask/cursedclown)
+	a_intent = INTENT_HARM
+	wander = FALSE
+	minimum_distance = 2
+	move_to_delay = 1
+	environment_smash = FALSE
+	pixel_y = 8
+	pressure_resistance = 200
+	minbodytemp = 0
+	maxbodytemp = 1500
+	atmos_requirements = list("min_oxy" = 0, "max_oxy" = 0, "min_tox" = 0, "max_tox" = 0, "min_co2" = 0, "max_co2" = 0, "min_n2" = 0, "max_n2" = 0)
+	var/mob/living/carbon/human/current_victim
+	var/manifested = FALSE
+	var/switch_stage = 60
+	var/stage = STAGE_HAUNT
+	var/interest = 0
+	var/target_area
+	var/invalid_area_typecache = list(/area/space, /area/centcom)
+	var/eating = FALSE
+	var/smiting = FALSE
+
+
+/mob/living/simple_animal/hostile/floor_cluwne/New()
+	. = ..()
+	var/obj/item/card/id/access_card = new (src)
+	access_card.access = get_all_accesses()//THERE IS NO ESCAPE
+	access_card.flags |= NODROP
+	invalid_area_typecache = typecacheof(invalid_area_typecache)
+	Manifest()
+	if(!current_victim)
+		Acquire_Victim()
+
+
+/mob/living/simple_animal/hostile/floor_cluwne/Destroy()
+	return ..()
+
+
+/mob/living/simple_animal/hostile/floor_cluwne/attack_hand(mob/living/carbon/human/M)
+	..()
+	playsound(src.loc, 'sound/items/bikehorn.ogg', 50, 1)
+
+
+/mob/living/simple_animal/hostile/floor_cluwne/CanPass(atom/A, turf/target)
+	return TRUE
+
+
+/mob/living/simple_animal/hostile/floor_cluwne/Life(seconds, times_fired)
+	do_jitter_animation(1000)
+	pixel_y = 8
+
+	if(is_type_in_typecache(get_area(src.loc), invalid_area_typecache))
+		var/area = pick(teleportlocs)
+		var/area/tp = teleportlocs[area]
+		forceMove(pick(get_area_turfs(tp.type)))
+
+	if(!current_victim)
+		Acquire_Victim()
+
+	if(stage && !manifested)
+		On_Stage()
+
+	if(stage == STAGE_ATTACK)
+		playsound(src, 'sound/spookoween/ghost_whisper.ogg', 75, 1)
+
+	if(eating)
+		return
+
+	var/turf/T = get_turf(current_victim)
+	if(prob(5))//checks roughly every 20 ticks
+		if(current_victim.stat == DEAD || current_victim.get_int_organ(/obj/item/organ/internal/honktumor/cursed) || is_type_in_typecache(get_area(T), invalid_area_typecache))
+			Acquire_Victim()
```

Comment:
Would it be better to put the `smiting` check inside `Acquire_Victim`? Otherwise, if the victim dies early or gets gibbed, the cluwne will go off and nab someone else. You'd want to move the `smiting` statement after `Acquire_Victim` first, of course.

Reply:
well i am useing the actual smite command to set the value..but i can also use acquire victim as a double check just in case they yeah died or gibbed before hand